### PR TITLE
Monorepo setup

### LINKED ISSUE #119: Add Strapi under version control
## To do

- [x] Refactor project structure
- [x] Setup Strapi
- [x] Create models
- [x] Add some test data
- [x] Create docker config
- [x] Add seed script
- [x] Migrate role permissions
- [x] Seed images
- [x] Seed product lists
- [x] Fix tests

## Notes

- The monorepo setup contains a docker compose file to spin up an instance of Strapi along with a PostgreSQL db. We could tie the strapi instance url to the name of the branch (e.g. `strapi-<BRANCH_NAME>.cominor.com` or `<BRANCH_NAME>.strapi.cominor.com`) so that it would be accessible by the vercel preview using `NEXT_PUBLIC_VERCEL_GIT_COMMIT_REF` env variable.

### PR DESCRIPTION
**Pull request by dhmacs**
This PR closes #119  

The purpose of this PR is to have Strapi config under version control, so that we can easily create multiple environments as well as support local development. I've discussed with @djmetzle how to setup the staging environments, and although ideally we would like to have as many staging environments as PRs that are open, we are taking a simpler approach for now. Once this PR is merged @djmetzle will setup a couple of staging environments, both based on the `main` branch.

## Changelog

- Moves root into `/frontend` folder
- Adds `/backend` folder with Strapi config files
- If `SEED_DB` env variable is `true` it seeds the database (useful to quickly bootstrap local development)
- Adds docker-compose file to boostrap Strapi server along with a Postgres db. This is used during CI to run integration tests now and can be used to deploy staging servers
- During CI tests run against a local instance of Strapi

## QA

Until we setup the staging servers we need to clone the repo and test it locally.

1. Clone this repo
2. Make sure you are running on node v14 (if you are using nvm run `nvm use` in the project working directory)
3. Install dependencies `npm install`
4. Rename `backend/.env.example` to `backend/.env` (this will enable db seed)
5. Run dev server: `npm run dev`
6. Make sure that everything compiles and the website runs correctly

### Access to local strapi instance

Email: `strapi@ifixit.com`
Password: `Password1`

### CODE REVIEW

**Review by djmetzle — APPROVED** [review]
CR :monorail:

**Review by evannoronha — APPROVED** [review]
CR 🔥 

Nice work @dhmacs. Makes sense and I'm excited to use and dev against this setup.

**Comment by danielbeardsley** [comment]
I'm pretty unfamiliar with strapi. To me, it's really hard to tell the difference between the auto-generated strapi config files (the output after creating collections and such) and stuff we may have hand edited / hand created it. I saw a lot of json and that made sense, but I also see components and executable javascript. Is that still part of the auto-generated output of strapi?

**Comment by dhmacs** [comment]
> I'm pretty unfamiliar with strapi. To me, it's really hard to tell the difference between the auto-generated strapi config files (the output after creating collections and such) and stuff we may have hand edited / hand created it. I saw a lot of json and that made sense, but I also see components and executable javascript. Is that still part of the auto-generated output of strapi?

@danielbeardsley the stuff I've worked on is under `/backend/db` and `/backend/config`, pretty much everything else has been generated by Strapi. During normal usage, if you have to create new content types, or update existing one, you just need to run the dev server (`npm run dev`) and open the strapi admin at `localhost:1337/admin` to make changes via the admin dashboard. The changes that you make will auto generate all the necessary config files, components, etc.
Having the config files under version control let us also customize Strapi, by editing existing files or creating new one.. We could for example create new components, new admin routes, new apis, but that's an advanced use case.

File: frontend/package.json [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      "concurrently": "6.2.1",
+      "cypress": "7.5.0",
+      "dotenv": "10.0.0",
+      "eslint": "7.19.0",
+      "eslint-config-next": "^11.0.1",
+      "eslint-config-prettier": "7.2.0",
+      "eslint-plugin-cypress": "2.11.3",
+      "eslint-plugin-react": "7.19.0",
+      "eslint-plugin-react-hooks": "4.2.0",
+      "husky": "4.2.3",
```
Comment (ardelato) [inline frontend/package.json:90]: Curious why version 4 is being used for Husky instead of version 7?

**Comment by ardelato** [comment]
QA 👍 

## Steps Taken:
1. Git cloned this repo locally
2. `cd` into the cloned repo
3. Checked out this branch
4. Made sure I was using the latest version of node 14 _(**14.17.6**)_ by using `nvm`
```
nvm install 14 && nvm use 14
```
5. I then ran npm install
```
npm install
```
6. I then renamed the env template
```
mv backend/.env.template backend/.env
```
7. After install the node packages I ran start
```
npm run start
```
8. I confirmed both the backend (http://localhost:1337) and frontend (http://localhost:3000) servers were started and successfully ran
9. I then tested the frontend pages:
 - [x] Confirmed the products had links to the store pages
 - [x] Confirmed querying worked
 - [x] Confirmed facet filtering worked
 - [x] Confirmed pagination works
 - [x] Confirmed reset filtering when no results found works
 - [x] Confirmed clear all facet filtering works
 - [x] Confirmed other collection pages worked 
10. I then tested the backend (strapi) by updating the description of the Parts page and confirmed it got updated in the frontend
Description updated
Strapi added description text

File: frontend/package.json (hunk 1 above)
Comment (dhmacs) [reply frontend/package.json:90]: @ardelato Hard to remember as this was done at the beginning of the project.. I think it was due to some breaking change with other tooling.. We can try to updated it later and see if it breaks something

**Pull request merged**